### PR TITLE
fix(cli): do not create migration script if repositories not selected

### LINKED ISSUE #7080: lb4 app - Fails on npm start when repository service is not chosen when creating
## Steps to reproduce

1. lb4 app
2. uncheck repositories
3. npm start

## Current Behavior
Receive error on migrate.ts
```
src/migrate.ts:9:13 - error TS2339: Property 'migrateSchema' does not exist on type 'MyAppApplication'.
9   await app.migrateSchema({existingSchema});
              ~~~~~~~~~~~~~
```

## Expected Behavior
1. app runs successfully with no additional change.

## Additional information

linux x64 10.23.2
+-- @loopback/boot@3.2.0
+-- @loopback/core@2.14.0
+-- @loopback/rest@9.1.2
+-- @loopback/rest-explorer@3.0.6
+-- @loopback/service-proxy@3.0.6

## Related Issues

Receiving same error as here: https://github.com/strongloop/loopback-next/issues/5608

_See [Reporting Issues](http://loopback.io/doc/en/contrib/Reporting-issues.html) for more tips on writing good issues_

### PR DESCRIPTION
**Pull request by pktippa**
Signed-off-by: Pradeep Kumar Tippa <pktippa@gmail.com>

Fixes #7080

Original PR was #7083

## Checklist

- [x] DCO (Developer Certificate of Origin) [signed in all commits](https://loopback.io/doc/en/contrib/code-contrib.html)
- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [x] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

**Review by achrinza — COMMENTED** [review]
Thanks for opening the pull request!

The CLI isn't really my in my domain, so I'll leave it to the other maintainers to approve;

I'm curious if we also need to update [`package.plain.json.ejs`](https://github.com/strongloop/loopback-next/blob/0bdd523470fa018b930e6cedd268c049f2ed3878/packages/cli/generators/project/templates/package.plain.json.ejs) and Project Generator as well.

I've also added a minor comment below, feel free to ignore it.

File: packages/cli/generators/app/index.js
Comment (pktippa) [reply packages/cli/generators/app/index.js:139]: @achrinza result is already shared at the end, do you mean we can delete this line?

Comment (achrinza) [reply packages/cli/generators/app/index.js:139]: > do you mean we can delete this line?

Yes

Comment (pktippa) [reply packages/cli/generators/app/index.js]: @achrinza changes done.

**Review by achrinza — APPROVED** [review]
Sorry for the delayed review, LGTM :+1: 

Please squash the PR into 1 commit; Then we'll be able to merge.

**Comment by achrinza** [comment]
I've rebased the changes from my end. I'd like one more approval (maybe from either @agnes512 or @emonddr) before merging the PR.

**Review by agnes512 — APPROVED** [review]
🚢

**Comment by dhmlau** [comment]
@achrinza @frbuceta, I was trying to land this PR but it failed at one of the GH action `Yuri6037/Action-FakeTTY@v1.1` but it doesn't show any error (thought it's coming from this line: https://github.com/strongloop/loopback-next/blob/master/.github/workflows/continuous-integration.yml#L42). Any idea what could go wrong? Thanks.

**Comment by frbuceta** [comment]
> @achrinza @frbuceta, I was trying to land this PR but it failed at one of the GH action `Yuri6037/Action-FakeTTY@v1.1` but it doesn't show any error (thought it's coming from this line: https://github.com/strongloop/loopback-next/blob/master/.github/workflows/continuous-integration.yml#L42). Any idea what could go wrong? Thanks.

I have been investigating the error with node v16 but locally it does not give any error. We must solve the problem with TTY and eliminate this extension, it does not give the option to see anything

**Comment by dhmlau** [comment]
@frbuceta, do you think it's safe to force merge it? I'm planning to publish a release some time this week. Thanks.

**Comment by frbuceta** [comment]
> @frbuceta, do you think it's safe to force merge it? I'm planning to publish a release some time this week. Thanks.

You have to override the PR and it should no longer give the error

**Comment by dhmlau** [comment]
@slnode test please

> You have to override the PR and it should no longer give the error

not sure if it works or need to push some changes in order to take effect.

**Comment by dhmlau** [comment]
@pktippa, thanks for your contribution. Your PR has landed! 🎉

**Pull request merged**
